jwt/src/main/scala/it/pagopa/interop/commons/jwt/service/impl/DefaultClientAssertionValidator.scala

```diff
@@ -27,20 +27,25 @@ trait DefaultClientAssertionValidator extends ClientAssertionValidator {
       _           = logger.debug("Getting subject claim")
       subject   <- subjectClaim(clientIdOpt, jwt.getJWTClaimsSet)
       purposeId <- Try(Option(jwt.getJWTClaimsSet.getStringClaim(PURPOSE_ID_CLAIM)))
-      kid       <- Try(jwt.getHeader.getKeyID)
+      kid       <- kidHeader(jwt)
     } yield ClientAssertionChecker(jwt, kid, subject, purposeId)
 
-  private def subjectClaim(clientId: Option[String], claimSet: JWTClaimsSet): Try[String] = {
-    Try(claimSet.getSubject) match {
-      case Failure(_) =>
-        logger.error("Subject not found in this claim")
+  private def subjectClaim(clientId: Option[String], claimSet: JWTClaimsSet): Try[String] =
+    Try(Option(claimSet.getSubject)) match {
+      case Failure(_) | Success(None)                     =>
+        logger.error("Subject not found in client assertion")
         Failure(SubjectNotFound)
-
-      case Success(s) if s == clientId.getOrElse(s) => Success(s)
-      case Success(s)                               =>
+      case Success(Some(s)) if s == clientId.getOrElse(s) => Success(s)
+      case Success(Some(s))                               =>
```

Comment:
[BIKESHEDDING ALERT] I would have flattened the 2 monads with something like `Option(claimSet.getSubject).toTry(SubjectNotFound).flatMap(Try)` keeping the same error handling